### ISSUE TITLE
nan output

### ISSUE THREAD

**Issue by superjax** [jizhang-cmu/demo_rgbd#21]
Thank you for making this package available as open source.  We have been very pleased with the performance of the visual odometry. However we are consistently experiencing the following unrecoverable bug:

The estimates are working fine when all of the sudden all of the features go red (no depth) and the odometry estimates become nan.  The depth images look fine when the dropout occurs.  This happens just about every time we use the code.  We saved a bagfile of images (magiccvs.byu.edu/vo_nan.bag) that highlights the problem.

**Comment by jizhang-cmu** [comment]
Please keep this dataset on the server. I'll find some time to look into the issue.

**Comment by pwnyholm** [comment]
Ji,

We found that in the imagePointsHandler function in visualOdometry.cpp, the ipr.s variable is set to NaN. We have not had a chance to track down whether it is a bad data point to begin with or if it is calculating a NaN, there are several places that ipr.s is set.

When ipr.s is NaN it sets the ipr3 and ipr4 x, y, and z members to NaN as well, on line 441, causing big problems.

We modified line 396 to look like this:
```
if (!std::isfinite(ipr.s)) {
    ipRelations->points[i].v = -1;
} else if (fabs(ipr.v) < 0.5) {
```
This hack seems to work (without much extensive testing), but it would be great to know why ipr.s is infinite in the first place and NaN check at the source.

Hopefully this helps point in the right direction, we'll keep you updated if we find anything else.

Thanks for your help
